esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -898,6 +898,28 @@ void WsWuInfo::getGraphTimingData(IArrayOf<IConstECLTimingData> &timingData)
         it->playProperties(visitor);
 }
 
+void WsWuInfo::getServiceNames(IEspECLWorkunit &info, unsigned long flags)
+{
+    if (!(flags & WUINFO_IncludeServiceNames))
+        return;
+
+    StringArray serviceNames;
+    WuScopeFilter filter("depth[1,]");
```

Comment:
You don't really need the depth filter - it will only filter a tiny fraction of the items, and will cost more to check.

For the filter you should be able to either use the string, or specify particular attributes.  You also shouldn't need to specify the source if you have provided enough information.

So you should either use
   WuScopeFilter filter("attr[servicename]");
or
   WuScopeFilter filter;
   filter.addOutputAttribute(WaServiceName);
   filter.finishedFilter();

Once the other PR has been merged this should be:
   WuScopeFilter filter("where[servicename],attr[servicename]");
or
   WuScopeFilter filter;
   filter.addOutputAttribute(WaServiceName);
   filter.addRequiredAttribute(WaServiceName);
   filter.finishedFilter();

There should also be a call to filter.finishedFilter();

Reply:
I thought that specifying a source will limit the search area. I will change.